judge/views/contests.py

```diff
@@ -62,20 +63,23 @@ def get_queryset(self):
         return Contest.get_visible_contests(self.request.user)
 
 
-class ContestList(DiggPaginatorMixin, TitleMixin, ContestListMixin, ListView):
+class ContestList(QueryStringSortMixin, DiggPaginatorMixin, TitleMixin, ContestListMixin, ListView):
     model = Contest
     paginate_by = 20
     template_name = 'contest/list.html'
     title = gettext_lazy('Contests')
     context_object_name = 'past_contests'
+    all_sorts = frozenset(('name', 'user_count', 'start_time'))
+    default_desc = frozenset(('name', 'user_count'))
+    default_sort = 'start_time'
 
     @cached_property
     def _now(self):
         return timezone.now()
 
     def _get_queryset(self):
         return super(ContestList, self).get_queryset() \
-            .order_by('-start_time', 'key').prefetch_related('tags', 'organizations', 'organizers')
+            .order_by(self.order, 'key').prefetch_related('tags', 'organizations', 'organizers')
```

Comment:
Since this `order_by` only affects past contests (and all other contests are sorted in Python), this `order_by` should probably be moved to `get_queryset` to save an uneeded `ORDER BY` in `get_context_data`.